app/src/main/java/org/mozilla/fenix/composables/TabTitle.kt

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.composables
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Default layout for a tab composable title.
+ *
+ * @param text tab title
+ * @param maxLines maximum number of lines for [text] to span, wrapping if necessary.
+ * If the text exceeds the given number of lines it will be ellipsized.
+ * @param modifier optional [Modifier] to be applied to the layout.
+ */
+@Composable
+fun TabTitle(
```

Comment:
These `TabTitle` and `TabSubtitle` functions might be better off living inside of `ListItemTabLarge` for the time being since the naming suggests it relates to a tab, and it's just a simple wrapper around `Text()`. At some point, you might want to call this `PrimaryText14` or even add a `fontSize` parameter. 

The way they are named now I would just expect people to just write their own `Text()` since a title might not be related to a tab, but a bookmark or history, etc. So, I could see us just grouping these composable functions to where they are currently used instead of separating them up completely.

Reply:
I have this outside of `ListItemTabLarge` and would expect everyone who needs to show a tab title to use this instead of using their own `Text()` to enforce the same style all throughout the app.
This is how a tab title should look like everywhere in the app.
This colors, fonts, etc.

Reply:
The flow I think of is that everytime we need to add a composable somewhere in the app we look for what already exists.
If it doesn't yet exist -> app design specs changed. Which means a big change and we should strive to avoid this.

With us just starting on this road, the "app design specs" are bound to change, to be added upon continuously.

But very important: if we want a cohesive design and reusable composables most of the time we'll have to extract a new composable which encomprises a new style/usecase to one that can be reused and avoid implementing it in place.

In this case, since we'd have multiple tab designs which would all need to show a title we can standardize how a tab title should look like and reuse that everywhere. Using a new Text() for this usecase and styling it in place would be a mistake.

Ideally all this reusable composables - their name and specs would come from UX who would have these in Figma and reuse them on new designs, with us doing the same in code. Then for both teams it would be easy to reuse existing designs and rethink the need for others.

Curious about your thoughts also @pocmo @mcarare @topotropic .

Reply:
IMO separating the composable and not tying them to their current usage place enforces reusability. This also makes sure the composable does not contain logic related to the current usage site. It also ensures a design change is reflected everywhere in the app when the same component is used. And it also makes generating previews for components easier. These are just initial thoughts, we can debate any one of them.